src/repo/fs.rs

```diff
@@ -115,6 +115,19 @@ impl BlockStore for FsBlockStore {
         }
         Ok(())
     }
+
+    async fn list(&self) -> Result<Vec<Result<Cid, Error>>, Error> {
+        let cids: Vec<Result<Cid, Error>> = self
+            .cids
+            .lock()
+            .unwrap()
+            .to_owned()
+            .iter()
+            .map(|cid| Ok(cid.to_owned()))
+            .collect();
```

Comment:
Here I think is happening:

* `to_owned()` is actually HashSet::clone, which makes this more difficult to see
* `iter()` is called on the cloned temporary, after which
* `Ok(cid.to_owned())` will again clone the values

What would suffice is:

```rust
// unwrapping as we want to panic on poisoned lock
let guard = self.cids.lock().unwrap();
guard.iter()
    .map(|cid| cid.clone())
    .collect()
```

This would turn the signature from `Result<Vec<Result<Cid, Error>>, Error>` to `Result<Vec<Cid>, Error>`, also I am not sure what can fail here? I guess in reality we would be iterating the directories so I think it's good to keep the outer error, but I can't see any reason for the inner?